rust/crd/src/affinity.rs

```diff
@@ -84,6 +84,8 @@ mod tests {
                                         )
                                     ]))
                                 }),
+                                match_label_keys: None,
```

Comment:
nit: Let's add a comment to these fields, something like: "Both these fields were added in Kubernetes 1.30, and cannot be used for now.".

Reply:
i defaulted since its a test https://github.com/stackabletech/hive-operator/pull/462/commits/590e178b2a39a28b63a6e385df4e16838798ab68